### DISCUSSION TITLE
Training a model in Vaex

### DISCUSSION THREAD

**Discussion by Som11-11** [vaexio/vaex#2054] [General]
Hello, I am new to Vaex library. I need assistance in understanding how the model is trained when we use Vaex.
Does it use observation one by one or does it copy the complete dataset in memory and train all at once.

**Comment by JovanVeljanoski** [comment]
That depends on the model, not on vaex.

Vaex does not implement ML models ( apart from KMeans means - which will work out of core). For the rest is up to the model: if scikit-learn, xgboost, lightgbm.. etc.. it will take the data and put it in memory as required by the models. 

If you use something like river, or incremental learning via sklearn, then it will use minibatches or update per sample.
